assets/js/app.js

```diff
@@ -10,70 +10,137 @@ import css from "../css/app.css"
+        '}'
+      ].join('\n'),
+      language: 'javascript',
+      //theme: 'vs-dark'
+    });
+    editorText = editor1.getValue();
+    editor1.getModel().onDidChangeContent((event) => {
+      if (editorText.localeCompare(editor1.getValue())) {
+        channel.push('shout', {
+          evnt: event,
+          text: editor1.getValue(),
+          updated_cursor_position: editor1.getPosition(),
+          //changed_by: user who made the change               
+        });
+      }
+    });
+    channel.on('shout', function (payload) {
+      var curpos = editor1.getPosition();
 
+      console.log(payload.evnt);
+      editorText = payload.text;
+      editor1.setValue(payload.text);
+      editor1.setPosition(curpos);
+      console.log(payload.curpos)
 
-// Since packaging is done by you, you need
-// to instruct the editor how you named the
-// bundles that contain the web workers.
-self.MonacoEnvironment = {
-  getWorkerUrl: function (moduleId, label) {
-    if (label === 'json') {
-      return './js/json.worker.js';
-    }
-    if (label === 'css') {
-      return './js/css.worker.js';
-    }
-    if (label === 'html') {
-      return './js/html.worker.js';
-    }
-    if (label === 'typescript' || label === 'javascript') {
-      return './js/ts.worker.js';
+      //for showing the cursor position of the user who made the change
+
+      var listOfCursors = []
+      // cursorPositionList[payload.changed_by]=payload.updated_cursor_position
+      // var style=document.createElement('style');
+      // style.type='text/css';
+      // document.getElementsByTagName('head')[0].appendChild(style);
+      // for(var i=0;i<noOfUsers;i++){
+      //   style.innerHTML='.cursor { background: '+userColors[i]+'; width:2px !important;;}';
+      //   listOfCursors.push({range: new monaco.Range(1,3,1,3), options: { className: 'cursor'}})
+      // }
+
+      var decorations = editor1.deltaDecorations([], listOfCursors);
+
+    });
+  }
+}
+
+class OnlineUsers {
+  displayUsers() {
+    function renderOnlineUsers(presence) {
+      let response = ""
+      presence.list((user, { metas: [first, ...rest] }) => {
+        let cursorColor = first["cursor_color"]
+        response += `<p style="color:${cursorColor};">${user}</p>`
+      })
+      let userList = document.getElementById("userList")
+      userList.innerHTML = response
     }
-    return './js/editor.worker.js';
-
-    // if (label === 'json') {
-    //     return './json.worker.bundle.js';
-    //   }
-    //   if (label === 'css') {
-    //     return './css.worker.bundle.js';
-    //   }
-    //   if (label === 'html') {
-    //     return './html.worker.bundle.js';
-    //   }
-    //   if (label === 'typescript' || label === 'javascript') {
-    //     return './ts.worker.bundle.js';
-    //   }
-    //   return './editor.worker.bundle.js';
+    presence.onSync(() => renderOnlineUsers(presence))
+    channel.join()
+  }
+}
+
+function generateColor() {
```

Comment:
This function can be replaced by a single line by knowing the max range of colors.

Reply:
Also, in the future, we'll probably want to have a set of colors which can be seen properly against the editors background instead of randomly assigning a color.